### PR TITLE
Changelog for 0.24.0

### PR DESCRIPTION
**Pull request by rgrunber**
CC @angelozerr @datho7561 

Signed-off-by: Roland Grunberg <rgrunber@redhat.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,29 @@
 # Change Log
 
+## [0.24.0](https://github.com/eclipse/lemminx/milestone/37?closed=1) (January 31, 2023)
+
+### Enhancements
+
+ * CodeLens, References, Rename, Diagnostics support for XML references. See [#1435](https://github.com/eclipse/lemminx/pull/1435), [#1427](https://github.com/eclipse/lemminx/issues/1427), [#1432](https://github.com/eclipse/lemminx/pull/1432), [#1452](https://github.com/eclipse/lemminx/pull/1452), [#1367](https://github.com/eclipse/lemminx/issues/1367).
```
Comment (datho7561) [inline CHANGELOG.md:7]: also linked editing, highlight, completion, definition

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,29 @@
 # Change Log
 
+## [0.24.0](https://github.com/eclipse/lemminx/milestone/37?closed=1) (January 31, 2023)
+
+### Enhancements
```
Comment (datho7561) [inline CHANGELOG.md:5]: - https://github.com/eclipse/lemminx/issues/1405 (it wasn't labelled correctly so it wasn't picked up)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,29 @@
...
+
+ * CodeLens, References, Rename, Diagnostics support for XML references. See [#1435](https://github.com/eclipse/lemminx/pull/1435), [#1427](https://github.com/eclipse/lemminx/issues/1427), [#1432](https://github.com/eclipse/lemminx/pull/1432), [#1452](https://github.com/eclipse/lemminx/pull/1452), [#1367](https://github.com/eclipse/lemminx/issues/1367).
+ * Code action to add missing required elements with RelaxNG. See [#1418](https://github.com/eclipse/lemminx/issues/1418).
+ * Improve the RelaxNG schema validation. See [#1425](https://github.com/eclipse/lemminx/pull/1425).
+ * Support for `textDocument/documentColor`. See [#639](https://github.com/eclipse/lemminx/issues/639).
+
+### Bug Fixes
```
Comment (datho7561) [inline CHANGELOG.md:12]: - https://github.com/redhat-developer/vscode-xml/issues/851 (submitted to vscode-xml but appiles to lemminx)

**Review by datho7561 — CHANGES_REQUESTED** [review]
Found a few things that were missing, otherwise looks good

File: CHANGELOG.md (hunk 3 above)
Comment (rgrunber) [reply CHANGELOG.md:12]: Should be https://github.com/eclipse/lemminx/pull/1433 then, but yes i'll add it.

**Review by datho7561 — APPROVED** [review]
Looks good to me. Thanks, Roland!

**Comment by angelozerr** [comment]
Thanks @rgrunber !

**Pull request merged**